### ISSUE TITLE
Account existence leaked when suspended through error messages

### ISSUE THREAD

**Issue by teknetia** [tootsuite/mastodon#16085]
Account existence, either current (suspended by a mod but not yet deleted) or past (deleted or suspension grace period has elapsed) is being leaked via error messages. This allows someone other than the account owner or instance admin to work out what state a profile may be in, leaking that information.

### Expected behaviour

All URLs to a profile that isn't visible should show a generic 404 error, such as the existing "The page you are looking for isn't here."

### Actual behaviour

- If the profile is currently suspended: "You don't have permission to view this page."
- If the profile was deleted by the user or suspension grace period has expired: "The page you were looking for doesn't exist here anymore."
- If the profile has never existed: "The page you are looking for isn't here."

**Comment by ClearlyClaire** [comment]
To be honest, I feel a suspended account should not be hidden information and returning a 404 on the generic HTML code path like we do rather than explicitly telling the account has been suspended, is an issue.

**Comment by teknetia** [comment]
I don't disagree when it comes to suspended accounts (especially considering #15324 is about making moderation decisions visible), but I do think deleted accounts at least should be treated the same as non-existent as there are many reasons someone may chose to leave an instance that would warrant privacy considerations (as far as they can be in a federated environment).

**Comment by ClearlyClaire** [comment]
I tend to agree, however for deleted accounts there is one technical decision that makes sense: a 410 Gone would be indicative of a deliberate decision to delete the question but 404 could be indicative of a temporary configuration failure. In particular, this matters to make decisions about deleting local data about a remote account: deleting on 410 is probably justified while deleting on 404 might be riskier.

(Mastodon doesn't have full automated pruning of accounts that have not been explicitly announced as deleted yet, but it's something I want to have eventually)

**Comment by ThisIsMissEm** [comment]
Is this potentially fixed by the split of Account deleted vs suspended? I know that was an issue for a while (or maybe still is?)

**Comment by ClearlyClaire** [comment]
That split has not happened yet (there are ways to tell an account that has self-deleted apart from an account that has been deleted after its suspension but those are brittle). Furthermore, this does not change the fact that we probably want to continue returning 410 for deleted accounts.

**Comment by ThisIsMissEm** [comment]
Okay, so this is probably closed as a "won't fix"?
